### PR TITLE
Draft For-User-Agent implementation

### PR DESCRIPTION
**Pull request by OStevan**
## Before this PR

In this draft tracing-java propagates an additional header across requests which will contain the user agent which was originally used to send a request.

Questions:
1. Should users of the library be able to set it when opening a span (essentially setting this info through the tracing lib)?
2. Should conjure clients set this header when sending a request?

_Tests are missing_

### CODE REVIEW

File: tracing-api/src/main/java/com/palantir/tracing/api/TraceHttpHeaders.java [hunk 1]
```diff
@@ -37,4 +37,6 @@
      */
     @Deprecated
     String ORIGINATING_SPAN_ID = "X-OrigSpanId";
+
+    String FOR_USER_AGENT = "X-For-User-Agent";
```
Comment (carterkozak) [inline tracing-api/src/main/java/com/palantir/tracing/api/TraceHttpHeaders.java:41]: nit: `X-` prefixes were deprecated several years ago in [RFC 6648](https://datatracker.ietf.org/doc/html/rfc6648)

File: tracing/src/main/java/com/palantir/tracing/Tracers.java [hunk 2]
```diff
@@ -48,6 +48,11 @@
      */
     public static final String REQUEST_ID_KEY = "_requestId";
 
+    /**
+     * The Key under which origin user agents are inserted into SLF4J {@link org.slf4j.MDC MDCs}.
+     */
+    public static final String FOR_USER_AGENT_KEY = "__forUserAgent";
```
Comment (carterkozak) [inline tracing/src/main/java/com/palantir/tracing/Tracers.java:54]: I don't think this belongs in the MDC, we can find it based on the requestId

**Review by carterkozak — COMMENTED** [review]
Took a quick scan over this (not terribly thorough). Before implementing this I think we should factor out common tracing state (information which applies to a trace, not individual spans) into a single object that's shared between Trace and DetachedSpan (e.g. traceId and requestId). That way features like this one don't require so much plumbing.

**Comment by OStevan** [comment]
Closing in favor of #790

**Pull request closed without merging**